awslambdaric/lambda_runtime_marshaller.py

```diff
@@ -8,14 +8,18 @@
 import simplejson as json
 
 from .lambda_runtime_exception import FaultException
+from .lambda_runtime_feature_handler import feature_handler
 
 
 # simplejson's Decimal encoding allows '-NaN' as an output, which is a parse error for json.loads
 # to get the good parts of Decimal support, we'll special-case NaN decimals and otherwise duplicate the encoding for decimals the same way simplejson does
 # We also set 'ensure_ascii=False' so that the encoded json contains unicode characters instead of unicode escape sequences
 class Encoder(json.JSONEncoder):
     def __init__(self):
-        super().__init__(use_decimal=False, ensure_ascii=False)
+        if feature_handler.is_feature_enabled("lambda_marshaller_ensure_ascii_false"):
```

Comment:
Q (not blocking): any particular reason to have feature gating inside the constructor and not having Encoder's class initializer parameterized? For visibility purposes might make sense to move the feature gating upwards of the call stack